### PR TITLE
allow -useprevious flag which merges any parameters in the existing

### PR DESCRIPTION
**Pull request by jpuffer**
Allow -useprevious flag which merges any parameters in the existing Cloud Formation stack that were not included in the ensure call

### CODE REVIEW

File: boondoggle/cli.py [hunk 1]
```diff
@@ -42,7 +42,9 @@ def cli(context, profile, region):
               help="Path to a cloudformation file.")
 @click.option('--outputs-from',
               help="Use another stack's outputs as inputs to this one.")
-def ensure(context, name, params, url, file, outputs_from):
+@click.option('--useprevious',
```
Comment (avram) [inline boondoggle/cli.py]: Look here: http://click.pocoo.org/5/options/#boolean-flags
```
@click.option('--shout/--no-shout', default=False)
```

Preserving the same semantics, we could do:
```
@click.option('--preserve/--no-preserve', default=False)
```

Or something. This resolves to a boolean

File: boondoggle/managers.py [hunk 2]
```diff
@@ -75,6 +79,21 @@ def ensure(self, name, parameters, url=None, path=None, outputs_from=None):
...
+
+    def fill_from_existing(self, new, old):
+        retParams = []
+        for oldParamObj in old:
+            found = False
+            for newTuple in new:
+                if newTuple[0] == oldParamObj.key:
+                    found = True
+                    retParams.append(newTuple)
+                    print('append new')
```
Comment (avram) [inline boondoggle/managers.py]: rm

File: boondoggle/cli.py (hunk 1 above)
Comment (jpuffer) [inline boondoggle/cli.py]: I don't see a use-case for --no-preserve so I think we can just add is_flag=True and use the option as a flag that you can exclude or include.  That way you don't have to pass True into it.

File: boondoggle/cli.py [hunk 3]
```diff
@@ -42,7 +42,9 @@ def cli(context, profile, region):
               help="Path to a cloudformation file.")
 @click.option('--outputs-from',
               help="Use another stack's outputs as inputs to this one.")
-def ensure(context, name, params, url, file, outputs_from):
+@click.option('--useprevious', is_flag=True,
```
Comment (avram) [inline boondoggle/cli.py]: This name isn't descriptive enough. `--preserve-parameters`?

File: boondoggle/cli.py [hunk 4]
```diff
@@ -42,7 +42,9 @@ def cli(context, profile, region):
...
+              help="Use existing stack's parameters if any are excluded from this call.")
```
Comment (avram) [inline boondoggle/cli.py]: "Default parameters to their current values, if any"?

**Comment by Raynes** [comment]
Suggestion not required for merge: run flake8 on this and then cry

File: boondoggle/cli.py [hunk 5]
```diff
@@ -42,7 +42,9 @@ def cli(context, profile, region):
               help="Path to a cloudformation file.")
 @click.option('--outputs-from',
               help="Use another stack's outputs as inputs to this one.")
-def ensure(context, name, params, url, file, outputs_from):
+@click.option('--use-previous-params', is_flag=True,
+              help="Default any excluded parameters to their existing values in Cloud Formation")
```
Comment (avram) [inline boondoggle/cli.py:46]: typo

Comment (jpuffer) [inline boondoggle/cli.py:46]: I don't see it.

Comment (avram) [inline boondoggle/cli.py:46]: CloudFormation is one word in AWS-speak

**Pull request merged**